source/kernel/src/test_util.rs

```diff
@@ -21,16 +21,25 @@ pub(crate) struct TestKernel {
 impl TestKernel {
     fn new() -> Self {
         trace_init();
+
+        // TODO(eliza): this clock implementation is also used in Melpomene, so
+        // it would be nice if we could share it with melpo...
+        let clock = {
+            maitake::time::Clock::new(Duration::from_micros(1), || {
+                SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_micros() as u64
```

Comment:
probably we could make this use `Instant` rather than `SystemTime`, but we'd need to stuff an initial start time `Instant` in a global so that this function can get at it. eventually i'd like to redo the `Clock` API to let you actually put data into the `Clock` at init...